bot/exts/moderation/incidents.py

```diff
@@ -146,6 +208,35 @@ async def add_signals(incident: discord.Message) -> None:
             await incident.add_reaction(signal_emoji.value)
 
 
+async def extract_message_links(message: discord.Message, bot: Bot) -> t.Optional[list]:
+    """
+    Checks if there's any message links in the text content.
+
+    Then passes the the message_link into `make_message_link_embed` to format a
+    embed for it containing information about the link.
+
+    As discord only allows a max of 10 embeds in a single webhook, just send the
+    first 10 embeds and don't care about the rest.
+
+    If no links are found for the message, it logs a trace statement.
+    """
+    message_links = DISCORD_MESSAGE_LINK_RE.findall(str(message.content))
```

Comment:
Does this need to convert to string?

Reply:
Which test is that?

Reply:
`ERROR: test_on_message_incident (tests.bot.exts.moderation.test_incidents.TestOnMessage)`

Reply:
It's failing because it doesn't mock the `content` attribute. So accessing it will resolve in a `MagicMock` instance. By converting it to a string in the tested code, you're running the regex on the following string:

```py
"<MagicMock name='mock.content' id='...'>"
```

Either way, I guess what I'm trying to ask is whether there's any reason to be converting the attribute to a string. [The docs](https://discordpy.readthedocs.io/en/stable/api.html#discord.Message.content) don't seem to think that it can be anything else. But even if it could, would it make sense to convert an arbitrary object into a string and be done with it?

By the way, the failing test is not testing this function directly. This code gets called by the tested function. The test does not make an attempt to assert what happens here, it only cares about whether the object is delegated to the correct functions under the correct conditions. Because the tested code was changed, the test's docstring is not entirely honest, since it does more than it's claiming to. I'm not saying that's necessarily wrong, just pointing it out.

Reply:
Uhh, I am not completely sure what you mean here.